contrib/win32/win32compat/misc.c

```diff
@@ -2160,3 +2175,42 @@ add_mark_of_web(const char* filename)
 	return status;
 }
 
+void get_zone_identifier(const char* hostname) {
+	static const CLSID CLSID_ISM =
+	{ 0x7B8A2D94, 0x0AC9, 0x11D1,
+	{ 0x89, 0x6C, 0x00, 0xC0, 0x4F, 0xB6, 0xBF, 0xC4 } };
+	static const IID IID_IISM =
+	{ 0x79EAC9EE, 0xBAF9, 0x11CE,
+	{ 0x8C, 0x82, 0x00, 0xAA, 0x00, 0x4B, 0xA9, 0x0B } };
+	IInternetSecurityManager* IISM;
+	CoInitialize(NULL);
+	HRESULT hr = CoCreateInstance(&CLSID_ISM, NULL,
+		CLSCTX_ALL, &IID_IISM, (void**)&IISM);
+	if (SUCCEEDED(hr) || hr == RPC_E_CHANGED_MODE)
+	{
+		wchar_t* hostname_w = utf8_to_utf16(hostname);
+		size_t inputStrLen = wcslen(hostname_w) + wcslen(L"ftp://") + 2;
+		wchar_t* inputStr = malloc(inputStrLen * sizeof(wchar_t));
+		if (inputStr == NULL) {
+			return;
+		}
+		swprintf_s(inputStr, inputStrLen, L"ftp://%s", hostname_w);
+		hr = IISM->lpVtbl->MapUrlToZone(IISM, inputStr, &motw_zone_id, 0);
+		if (hr == S_OK) {
```

Comment:
```suggestion
		if (hr == S_OK) {
		    debug("MapUrlToZone zone identifier value: %d", motw_zone_id);
```
We can then remove the `if (motw_zone_id < 5) { ...`, but still detect and report `unexpected` value.